docs/self-hosted-appcircle/install-server/helm-chart/configuration/production-readiness/gcp-cloud-storage-configuration.md

```diff
@@ -0,0 +1,634 @@
+---
+title: GCP Cloud Storage Configuration
+description: Learn how to configure GCP Cloud Storage as object storage for Appcircle Server when installing using a Helm chart
+tags:
+  [
+    self-hosted,
+    gcp,
+    cloud-storage,
+    object-storage,
+  ]
+sidebar_position: 60
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import NeedHelp from '@site/docs/_need-help.mdx';
+import S3MinimumVersionCaution from '@site/docs/self-hosted-appcircle/install-server/helm-chart/configuration/\_s3-minimum-version-caution.mdx';
+
+## Overview
+
+This guide provides comprehensive instructions for configuring **Google Cloud Storage (GCS)** as your object storage backend for the Appcircle Server. While the default Helm chart deployment includes MinIO as an in-cluster object storage solution, **production environments** benefit from using a more robust and scalable solution like GCP Cloud Storage.
+
+<S3MinimumVersionCaution />
+
+### What This Guide Covers
+
+This guide will walk you through the process of configuring GCP Cloud Storage as your object storage backend for the Appcircle Server. To use GCP Cloud Storage with Appcircle Server, you need to:
+
+- **Set up GCP infrastructure**: GCS buckets, IAM service accounts, and permissions
+- **Configure Appcircle Server**: Update Helm values to use GCS
+- **Optional CDN setup**: Google Cloud CDN for performance optimization
+
+:::caution
+GCP Cloud Storage configuration requires a **fresh installation** of Appcircle Server. If you have an existing installation with data, migration from MinIO to GCP Cloud Storage is **not supported**. You can only configure GCP Cloud Storage during the initial installation or if you have no existing data to preserve.
```

Comment:
1️⃣ This should be a general caution for other S3 configurations as well. So we should include it not only here but also in other S3 configuration pages since we do not have any data migration steps to take for the configured S3.

2️⃣ We should not mention specific object storage in the statement below:
> ... migration from MinIO to GCP Cloud Storage ...

Instead we can use "... migration for the previously configured object storage to GCP ...".